scipy/sparse/tests/test_base.py

```diff
@@ -4300,7 +4300,10 @@ def spmatrix(self, D, **kwargs):
                                            zero_pos[0][k],
                                            zero_pos[1][k])
 
-        arg1 = self._arg1_for_noncanonical(M)
+        if sorted_indices:
+            arg1 = self._arg1_for_noncanonical(M, sorted_indices)
+        else:
+            arg1 = self._arg1_for_noncanonical(M)
```

Comment:
Alright, will change that. I'm not sure about BSR, what do you think should be tested? It has no `get_element`. Testing everything with sorted and unsorted indices leads us back to my previous commit (too many tests).